src/adapter/activateDaffodilDebug.ts

```diff
@@ -188,13 +192,11 @@ async function createDebugRunFileConfigs(
         request: 'launch',
         type: 'dfdl',
         schema: {
-          path: tdmlConfig?.action === 'execute' ? '' : normalizedResource,
+          ...(newSchema && { path: newSchema }),
           rootName: null,
           rootNamespace: null,
         },
-        data:
-          tdmlConfig?.action === 'execute' ? '' : '${command:AskForDataName}',
-        debugServer: false,
+        ...(newData && { data: newData }),
```

Comment:
You don't need to do this, it can be simplified to:

```suggestion
        data: newData,
```